sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_models.py

```diff
@@ -351,3 +357,114 @@ def _to_generated(self) -> KeyValue:
+     Required.
+    :vartype filters: list[dict[str, str]]
+    :ivar composition_type: The composition type describes how the key-values within the snapshot
+     are composed. The 'key' composition type ensures there are no two key-values containing the
+     same key. The 'key_label' composition type ensures there are no two key-values containing the
+     same key and label. Known values are: "key" and "key_label".
+    :vartype composition_type: str
+    :ivar created: The time that the snapshot was created.
+    :vartype created: ~datetime.datetime
+    :ivar expires: The time that the snapshot will expire.
+    :vartype expires: ~datetime.datetime
+    :ivar retention_period: The amount of time, in seconds, that a snapshot will remain in the
+     archived state before expiring. This property is only writable during the creation of a
+     snapshot. If not specified, the default lifetime of key-value revisions will be used.
+    :vartype retention_period: int
+    :ivar size: The size in bytes of the snapshot.
+    :vartype size: int
+    :ivar items_count: The amount of key-values in the snapshot.
+    :vartype items_count: int
+    :ivar tags: The tags of the snapshot.
+    :vartype tags: dict[str, str]
+    :ivar etag: A value representing the current state of the snapshot.
+    :vartype etag: str
+    """
+    
+    def __init__(
+        self,
+        *,
+        filters: List[Dict[str, str]],
+        composition_type: Optional[Literal["key", "key_label"]] = None,
+        retention_period: Optional[int] = None,
+        tags: Optional[Dict[str, str]] = None,
+        ** kwargs
+    ):
+        """
+        :keyword filters: A list of filters used to filter the key-values included in the snapshot.
+         Required.
+        :paramtype filters: list[dict[str, str]]
+        :keyword composition_type: The composition type describes how the key-values within the
+         snapshot are composed. The 'key' composition type ensures there are no two key-values
+         containing the same key. The 'key_label' composition type ensures there are no two key-values
+         containing the same key and label. Known values are: "key" and "key_label".
+        :paramtype composition_type: str
+        :keyword retention_period: The amount of time, in seconds, that a snapshot will remain in the
+         archived state before expiring. This property is only writable during the creation of a
+         snapshot. If not specified, the default lifetime of key-value revisions will be used.
+        :paramtype retention_period: int
+        :keyword tags: The tags of the snapshot.
+        :paramtype tags: dict[str, str]
+        """
+        self.name = kwargs.get("name", None)
+        self.status = kwargs.get("status", None)
+        self.filters = filters
+        self.composition_type = composition_type
+        self.created = kwargs.get("created", None)
```

Comment:
Why are we setting all the readonly properties from kwargs?
I'm not sure this is the best way to do this - if it's only to support the `_from_generated` method, we can set them outside the constructor.